src/main/java/net/minecraftforge/fluids/FluidAttributes.java

```diff
@@ -291,6 +287,11 @@ public int getColor()
         return color;
     }
 
+    public boolean getUseBiomeWaterColor()
```

Comment:
This flag is too specific to be here, it is not at the right level of abstraction.

Instead, I think we should create subclass of `FluidAttributes` just for water that overrides `FluidAttributes#getColor(IEnviromentBlockReader, BlockPos)` to call `BiomeColors.getWaterColor`.
That should simplify the patch a little as well.

Reply:
I was initially thinking I don't like the idea, but actually, after poking at it in the code, I do.

It actually doesn't change the patch that much, but I guess the logic in FluidBlockRenderer is simpler now.